airflow/executors/kubernetes_executor.py

```diff
@@ -315,6 +308,21 @@ def _run(self,
 
         return last_resource_version
 
+    def _get_watcher_args(self,
+                          resource_version: Optional[str],
+                          worker_uuid: str,
+                          kube_config: Any) -> dict:
+        """Builds and returns the kwargs necessary for Watcher"""
+        kwargs = {'label_selector': 'airflow-worker={}'.format(worker_uuid)}
+        if resource_version:
+            kwargs['resource_version'] = resource_version
+        if kube_config.kube_client_request_args:
+            for key, value in kube_config.kube_client_request_args.iteritems():
+                kwargs[key] = value
+            conn_timeout = kube_config.kube_client_request_args.get('_request_timeout', [60, 60])[0]
+            kwargs['timeout_seconds'] = conn_timeout - 1 if conn_timeout - 1 > 0 else 1
```

Comment:
```suggestion
            kwargs['timeout_seconds'] = max(conn_timeout - 1, 1)
```

(Assuming they are integers.)